sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -83,23 +83,29 @@ static RT::PiProgram createSpirvProgram(const RT::PiContext Context,
   return Program;
 }
 
-RT::PiProgram ProgramManager::getBuiltOpenCLProgram(OSModuleHandle M,
-                                                    const context &Context) {
+RT::PiProgram
+ProgramManager::getBuiltOpenCLProgram(OSModuleHandle M, const context &Context,
+                                      const string_class &KernelName) {
   std::shared_ptr<context_impl> Ctx = getSyclObjImpl(Context);
-  std::map<OSModuleHandle, RT::PiProgram> &CachedPrograms =
+  std::map<OSModuleHandle, std::vector<RT::PiProgram>> &CachedPrograms =
```

Comment:
I'd suggest sparing a `using` directive for `std::vector<RT:::PiProgram>`, as it's used in a couple places around the code

Reply:
I don't think `std::vector<RT::PiProgram>` is verbose enough to warrant that, IMO introducing an alias for such a succinct template might even harm the readability by adding unnecessary obfuscation.

Reply:
Fine with me, resolved